src/datatypes/src/vectors/primitive.rs

```diff
@@ -142,8 +148,11 @@ impl<T: LogicalPrimitiveType> PrimitiveVector<T> {
                     .to_data(),
                 _ => unreachable!(),
```

Comment:
question: Not sure whether other units are possible to use. Is it really unreachable? @v0y4g3r

Reply:
@evenyag May be I can answer this question. 
In my initial version,I modify all `unreachable!()` to `CastTypeSnafu`, but when I want to test this logic branch,
I find that `arrow-array` only support `Time32SecondArray`,`Time32MillisecondArray`,`Time64MicrosecondArray` and `Time64NanosecondArray`.  that means we can't get like `Time32MicrosecondArray` or `Time64SecondArray` and so on. So I think it's unreachable.